### PR TITLE
Some random coverage improvements for DynamoCore assembly

### PR DESCRIPTION
**Pull request by ellensi**
### Purpose
- Modified a test case to cover `Dynamo.Search.SearchCategoryUtil.GetAllCategoryNames()`
- Removed a private constructor `Dynamo.Graph.Presets.PresetModel(string, string, List<NodeModel>, List<XmlElement>, Guid)`. This constructor was once used for the deserialization method, which by now had been changed to have its own implementation for this purpose.
- Added a test case that covers `Dynamo.Search.NodeSearchModel.DumpLibraryToXml(string, string)`, `Dynamo.Models.DynamoModel.DumpLibraryToXml()` and `Dynamo.Models.DynamoModel.CanDumpLibraryToXml()`
### Declarations
- [x] The code base is in a better state after this PR
- [x] The level of testing this PR includes is appropriate
- [x] All tests pass using the self-service CI.
### Reviewers

@monikaprabhu

### CODE REVIEW

File: test/DynamoCoreWpfTests/CoreUITests.cs [hunk 1]
```diff
@@ -797,6 +797,32 @@ public void WorkspaceContextMenu_IfSubmenuOpenOnMouseHover()
...
+        [Category("UnitTests")]
+        public void TestDumpLibraryToXml()
+        {
+            ViewModel.DumpLibraryToXmlCommand.Execute(null);
+
+            // Look for library dump files aged within 10 seconds from the current time
+            var time = System.DateTime.Now;
+            var timeout = 10;
+
+            for (var t = 0; t < timeout; t++)
```
Comment (Benglin) [inline test/DynamoCoreWpfTests/CoreUITests.cs]: I'd write this as `for (var attempt = 0; attempt < attempts; ++attempt)`, just a small suggestion.

File: test/DynamoCoreWpfTests/CoreUITests.cs [hunk 2]
```diff
@@ -797,6 +797,32 @@ public void WorkspaceContextMenu_IfSubmenuOpenOnMouseHover()
...
+            {
+                string fileName = string.Format("LibrarySnapshot_{0}.xml", time.ToString("yyyyMMddHmmss"));
+                string fullFileName = Path.Combine(Model.PathManager.LogDirectory, fileName);
+                Console.WriteLine("Looking for library dump file at " + fullFileName);
+                if (File.Exists(fullFileName))
+                {
+                    File.Delete(fullFileName);
```
Comment (Benglin) [inline test/DynamoCoreWpfTests/CoreUITests.cs]: Can we redirect the XML file to the folder indicated by `SystemTestBase.TempFolder`? I believe that folder is randomly generated and will always be destroyed towards the end of the test. My point is, we may not even need to delete the output file manually here.

Comment (ellensi) [inline test/DynamoCoreWpfTests/CoreUITests.cs]: The (parameterless) command should actually write into this location, not a temporary folder:
`%APPDATA%/Dynamo/Dynamo Core/1.0/Logs/LibrarySnapshot_20160615235959.xml`

We might not be able to specify a different output folder for the test case in a straightforward manner, but I'm good to go if you feel that we don't need to delete the file after running the test case.

File: test/DynamoCoreWpfTests/CoreUITests.cs [hunk 3]
```diff
@@ -797,6 +797,32 @@ public void WorkspaceContextMenu_IfSubmenuOpenOnMouseHover()
             Assert.IsTrue(currentWs.WorkspaceLacingMenu.IsSubmenuOpen);
         }
 
+        [Test]
+        [Category("UnitTests")]
+        public void TestDumpLibraryToXml()
+        {
+            ViewModel.DumpLibraryToXmlCommand.Execute(null);
+
```
Comment (sharadkjaiswal) [inline test/DynamoCoreWpfTests/CoreUITests.cs]: I think this test is not to test whether DumpLibrary.... method is callable. You must validate the output against existing data. For example, here you can verify if the XML file is created or not, and then how many elements are created is it same as the number of items in the library view? 

There should be a way to know the file name dumped, can we have a command completed event with output path as a parameter, so you don't need to iterate and guess the xml file name.

**Comment by QilongTang** [comment]
Ping @ellensi

**Comment by Racel** [comment]
@ellensi @QilongTang @Benglin @sharadkjaiswal - It doesn't look like anyone is working on this? Can we close? Can you checkout today? I want to ensure our backlog of PRs is up-to-date for the start of next sprint which begins on Wednesday.

**Comment by QilongTang** [comment]
@Racel @ramramps @mjkkirschner I am closing this PR since reading all the comments above indicating this PR is not ready to be merge and needs further work

**Pull request closed without merging**